src/Symfony/Component/Cache/DependencyInjection/CacheCollectorPass.php

```diff
@@ -46,29 +46,41 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        $collectorDefinition = $container->getDefinition($this->dataCollectorCacheId);
         foreach ($container->findTaggedServiceIds($this->cachePoolTag) as $id => $attributes) {
-            $definition = $container->getDefinition($id);
-            if ($definition->isAbstract()) {
-                continue;
-            }
+            $this->addToCollector($id, $container);
```

Comment:
don't we want to collect only the decorator when there is one?

Reply:
No, If we only collect the TagAwareAdapter, we lose the psr6 calls since the Adapter decorated by TagAwareAdapter calls these methods on itself.